src/OrchardCore.Modules/OrchardCore.Forms/Controllers/ValidationRuleController.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+using OrchardCore.ContentManagement;
+using OrchardCore.Forms.Helpers;
+using OrchardCore.Forms.Models;
+using OrchardCore.Forms.Services;
+using OrchardCore.Forms.Services.Models;
+
+namespace OrchardCore.Forms.Controllers
+{
+    [Route("api/validationApi")]
+    [Authorize(AuthenticationSchemes = "Api"), IgnoreAntiforgeryToken, AllowAnonymous]
+    [ApiController]
+    public class ValidationApiController : Controller
+    {
+        private readonly IAuthorizationService _authorizationService;
+        private readonly ValidationRuleHelpers _validationRuleHelpers;
+        private readonly IStringLocalizer S;
+        public ValidationApiController(
+            IAuthorizationService authorizationService,
+            ValidationRuleHelpers validationRuleHelpers,
+            IStringLocalizer<ValidationApiController> stringLocalizer)
+        {
+            _authorizationService = authorizationService;
+            _validationRuleHelpers = validationRuleHelpers;
+            S = stringLocalizer;
+        }
+        [HttpGet]
+        [Route("ValidateFormByRule")]
+        public async Task<IActionResult> ValidateFormByRule(string contentItemId, string formName, string formValue)
```

Comment:
Isn't it the opposite of what was advertised in the first place, and the form is validated server-side all the time? Or is it just getting a field of the form and not the whole form.

Reply:
both of form filed and  form will be validated by this function, every form field will request that api for validation before client form submit.